CloudOnMobile/CommandManager.swift

```diff
@@ -45,25 +23,44 @@ final class CommandManager {
         var payload: T?
     }
 
-    func connect() {
-        Thread.detachNewThread {
+    private var code: UInt32 = 0
+
+    private var isConnected = false
+
+    private let ip: String
+
+    private let port: Int = 9293
+
+    private let documentsDirectory: URL
+
+    init(url: String) {
+        ip = url
+        initSharedCore()
+        let fileMngr = FileManager.default
+        documentsDirectory = fileMngr.urls(for: .documentDirectory, in: .userDomainMask)[0]
+        setup_environment(documentsDirectory.path)
+    }
+
+    func connect() async throws -> Int {
+        try await withUnsafeThrowingContinuation { continuation in
             self.ip.withCString { (ip_cstr: UnsafePointer<Int8>) in
-                print("Start tcp-client thread")
                 let code = UnsafeMutablePointer<UInt32>.allocate(capacity: 1)
-                if connect_to_server(ip_cstr, Int32(self.port), code) != 0 {
+                guard connect_to_server(ip_cstr, Int32(self.port), code) == 0 else {
                     self.isConnected = false
-                    self.delegate?.serverOnClose() // TODO: connection failure
+                    continuation.resume(throwing: CommandManagerError.connectionFailure)
                     return
                 }
                 self.isConnected = true
-                self.delegate?.serverOnConnected(passocde: Int(code.pointee))
+                let value = Int(code.pointee)
+                continuation.resume(returning: value)
                 code.deallocate()
-              _ = self.endlessListen()
+                _ = self.endlessListen()
```

Comment:
would it be cleaner to make this function return immediately and just return a status with the delegate pattern? E.g. `self.delegate?.serverOnClose()`, `onConnected()`?